src/client.rs

```diff
@@ -114,14 +113,14 @@ impl Client {
     }
 
     /// Builds a request
-    pub fn build_request(&self, name: String, params: Vec<serde_json::Value>) -> Request {
+    pub fn build_request<'a>(&self, name: &'a str, params: &'a [serde_json::Value]) -> Request<'a> {
```

Comment:
Can you split this into two lifetimes, one for `name` and one for `params`?

Reply:
Done.